packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -1220,6 +1378,47 @@ class MemoryTimeline {
   }
 }
 
+/// Compute a simple moving average.
+class MovingAverage {
+  MovingAverage({this.averagePeriod = 20, this.magnitudeSpike = 3 });
+
+  final dataSet = Queue<int>();
+
+  // Total collected items in the X axis (time) used to compute moving average.
+  final int averagePeriod;    // Default 10-20 seconds.
+
+  final int magnitudeSpike;   // Default magnitude of comparing first to mean.
+
+  // Sum of total heap used and external heap for unitPeriod.
+  int averageSum = 0;
+
+  // Reset moving average data.
+  void clear() {
+    dataSet.clear();
+    averageSum = 0;
+  }
+
+  // Update the sum to get a new mean.
+  void addToMA(int value) {
+    averageSum += value;
+    dataSet.add(value);
+
+    // Update dataSet of values to not exceede the period of the moving average
+    // to compute the normal mean.
+    if (dataSet.length > averagePeriod) {
+      averageSum -= dataSet.removeFirst();
+    }
+  }
+
+  int mean() => averageSum ~/ min(averagePeriod, dataSet.length);
```

Comment:
change this to a getter and handle `dataSet.isEmpty` robustly. zero is probably a better result to return for that case than `NaN`